roles/tackle/tasks/main.yml

```
@@ -1,319 +1,300 @@
+  set_fact:
+    openshift_cluster: true
+  when: "'route.openshift.io' in api_groups"
+
+- when: openshift_cluster|bool
+  block:
+    - name: "Get OpenShift cluster Proxy object"
+      set_fact:
+        proxy_cluster: "{{ lookup('k8s', api_version='config.openshift.io/v1', kind='Proxy', resource_name='cluster') }}"
+
+    - when: (proxy_cluster.spec.trustedCA.name | length) > 0
+      block:
+        - name: "Enable trusted CA environment"
+          set_fact:
+            trusted_ca_enabled: true
+
+        - name: "Create an empty ConfigMap that will hold the trusted CA"
+          k8s:
+            state: present
+            definition: "{{ lookup('template', 'configmap-trusted-ca.yml.j2') }}"
+
+- when:
+    - feature_auth_required|bool
+  block:
+    - name: "Setup Keycloak PostgreSQL PersistentVolumeClaim"
+      k8s:
+        state: present
+        definition: "{{ lookup('template', 'persistentvolumeclaim-keycloak-postgresql.yml.j2') }}"
+
+    - name: "Check if Keycloak PostgreSQL Secret exists already so we don't update it"
+      k8s_info:
+        api_version: v1
+        kind: Secret
+        name: "{{ keycloak_database_secret_name }}"
+        namespace: "{{ app_namespace }}"
+      register: keycloak_database_secret_status
+
+    - when: (keycloak_database_secret_status.resources | length) == 0
+      block:
+        - name: "Generate random values for Keycloak PostgreSQL credentials"
+          set_fact:
+            keycloak_database_db_username: "user-{{ lookup('password', '/dev/null chars=ascii_lowercase,ascii_uppercase,digits length=4') }}"
+            keycloak_database_db_password: "{{ lookup('password', '/dev/null chars=ascii_lowercase,ascii_uppercase,digits length=16') }}"
+
+        - name: "Encode Keycloak PostgreSQL credentials"
+          set_fact:
+            keycloak_database_db_username_b64: "{{ keycloak_database_db_username | b64encode }}"
+            keycloak_database_db_password_b64: "{{ keycloak_database_db_password | b64encode }}"
+
+        - name: "Setup Keycloak PostgreSQL Secret"
+          k8s:
+            state: present
+            definition: "{{ lookup('template', 'secret-keycloak-postgresql.yml.j2') }}"
+
+    - name: "Setup Keycloak PostgreSQL Service"
+      k8s:
+        state: present
+        definition: "{{ lookup('template', 'service-keycloak-postgresql.yml.j2') }}"
+
+    - name: "Setup Keycloak PostgreSQL Deployment"
+      k8s:
+        state: present
+        definition: "{{ lookup('template', 'deployment-keycloak-postgresql.yml.j2') }}"
+
+    - name: "Check status of Keycloak PostgreSQL Deployment"
+      k8s_info:
+        api_version: v1
+        kind: Pod
+        namespace: "{{ app_namespace }}"
+        label_selectors:
+           - app.kubernetes.io/name = {{ keycloak_database_service_name }}
+      register: pod
+      until: pod.resources[0].status.phase == "Running"
+      retries: 20
+      delay: 2
+      ignore_errors: true
```

An extra space on `- app.kubernetes.io/name = {{ keycloak_database_service_name }}`?

```suggestion
    - name: "Check status of Keycloak PostgreSQL Deployment"
      k8s_info:
        api_version: v1
        kind: Pod
        namespace: "{{ app_namespace }}"
        label_selectors:
          - app.kubernetes.io/name = {{ keycloak_database_service_name }}
      register: pod
      until: pod.resources[0].status.phase == "Running"
      retries: 20
      delay: 2
      ignore_errors: true
```